server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/draft/model/message/view/MessageViewFactory.java

```diff
@@ -171,5 +177,44 @@ static String getNameOrAddress(Mailbox mailbox) {
                 .flatMap(toMessageViews(converter))
                 .collect(Guavate.toImmutableList());
         }
+
+        static Instant getDateFromHeaderOrInternalDateOtherwise(Message mimeMessage, MessageResult message) {
+            return Optional.ofNullable(mimeMessage.getDate())
+                .map(Date::toInstant)
+                .orElse(message.getInternalDate().toInstant());
+        }
+
+        static Instant getDateFromHeaderOrInternalDateOtherwise(Message mimeMessage, MessageFullViewFactory.MetaDataWithContent message) {
+            return Optional.ofNullable(mimeMessage.getDate())
+                .map(Date::toInstant)
+                .orElse(message.getInternalDate());
+        }
+
+
+        static Message parse(MessageResult message) throws MailboxException {
+            try {
+                return Message.Builder
+                    .of()
+                    .use(MimeConfig.PERMISSIVE)
+                    .parse(message.getFullContent().getInputStream())
+                    .build();
+            } catch (IOException e) {
+                throw new MailboxException("Unable to parse message: " + e.getMessage(), e);
+            }
+        }
+
+
+        static Message parse(MessageFullViewFactory.MetaDataWithContent message) throws MailboxException {
```

Comment:
Add a new common `static Message parse(InputStream message) throws MailboxException` so that we do not repeat ourselves above...